pkg/apiserver/authenticator/authn.go

```diff
@@ -96,12 +99,20 @@ func New(config AuthenticatorConfig) (authenticator.Request, error) {
 		authenticators = append(authenticators, serviceAccountAuth)
 	}
 
-	if len(config.KeystoneURL) > 0 {
-		keystoneAuth, err := newAuthenticatorFromKeystoneURL(config.KeystoneURL)
-		if err != nil {
-			return nil, err
+	if len(config.KeystoneConfig) > 0 {
```

Comment:
I don't see why password and token auth are mutually exclusive here... I'd sort of expect to be able to set up keystone basic auth (with just the URL, as you can today?) and/or keystone token auth (by providing a config file like this?)